datadog/threadstats/base.py

```diff
@@ -146,7 +146,7 @@ def event(self, title, text, alert_type=None, aggregation_key=None,
                 source_type_name=source_type_name, date_happened=date_happened, priority=priority,
                 tags=event_tags, host=hostname)
 
-    def gauge(self, metric_name, value, timestamp=None, tags=None, sample_rate=1, host=None):
+    def gauge(self, metric, value, timestamp=None, tags=None, sample_rate=1, host=None):
```

Comment:
```suggestion
    def gauge(self, metric=None, value=None, timestamp=None, tags=None, sample_rate=1, host=None, metric_name=None):
```
- [ ] `raise TypeError` when `value is None` or `(metric or metric_name) is None`